_studio/enctools/src/mfx_enctools_brc.cpp

```diff
@@ -894,6 +894,23 @@ mfxStatus BRC_EncTool::UpdateFrame(mfxU32 dispOrder, mfxEncToolsBRCStatus *pFram
     return sts;
 }
 
+mfxStatus BRC_EncTool::GetHRDPos(mfxU32 dispOrder, mfxEncToolsBRCHRDPos *pHRDPos)
+{
+    MFX_CHECK(m_bInit, MFX_ERR_NOT_INITIALIZED);
+    MFX_CHECK_NULL_PTR1(pHRDPos);
+
+    auto frameStructItr = std::find_if(m_FrameStruct.begin(), m_FrameStruct.end(), CompareByDisplayOrder(dispOrder));
+    if (frameStructItr == m_FrameStruct.end())
+        return MFX_ERR_UNDEFINED_BEHAVIOR; // BRC hasn't processed the frame
+    BRC_FrameStruct frameStruct = *frameStructItr;
```

Comment:
Only frameStructItr->encOrder is needed. No need to create an then copy the structure.

Reply:
thank you fixed